src/common/main/kotlin/tech/thatgravyboat/skyblockapi/api/remote/api/SimpleItemAPI.kt

```diff
@@ -72,45 +91,72 @@ object SimpleItemAPI {
     }
 
     fun getRuneByIdOrNull(id: SkyBlockId): ItemStack? = cache.getOrPut(id.trySafe(::rune)) {
-        val runeId = id.toApiIdOrNull() ?: return@getOrPut null
+        val runeId = id.cleanId.uppercase()
+
+        if (runeId == UNKNOWN) {
+            return@getOrPut null
+        }
 
         if (runeId.contains(":")) {
             val (runeId, literalLevel) = runeId.split(":")
             val level = literalLevel.toIntValue()
             RepoRunesAPI.getRuneAsItemOrNull(runeId, level)?.let { return@getOrPut it }
         }
 
-        RepoRunesAPI.getRuneAsItemOrNull(runeId.substringBefore(":"))
+        for (i in 3 downTo 0) {
+            RepoRunesAPI.getRuneAsItemOrNull(runeId.substringBefore(":"), i)?.let { return@getOrPut it }
+        }
+
+        return@getOrPut null
     }
 
     fun getRuneById(id: SkyBlockId) = getRuneByIdOrNull(id) ?: ItemBuilder(Items.BARRIER) {
         name("Unknown rune: $id")
     }
 
     fun getEnchantmentByIdOrNull(id: SkyBlockId): ItemStack? = cache.getOrPut(id.trySafe(::enchantment)) {
-        val enchantmentId = id.toApiIdOrNull() ?: return@getOrPut null
+        val enchantmentId = id.cleanId.uppercase()
+
+        if (enchantmentId == UNKNOWN) {
+            return@getOrPut null
+        }
 
         if (enchantmentId.contains(":")) {
             val (enchantmentId, literalLevel) = enchantmentId.split(":")
             val level = literalLevel.toIntValue()
             RepoEnchantmentAPI.getEnchantmentAsItemOrNull(enchantmentId, level)?.let { return@getOrPut it }
         }
 
-        RepoEnchantmentAPI.getEnchantmentAsItemOrNull(enchantmentId.substringBefore(":"))
+        for (i in 10 downTo 0) {
+            RepoEnchantmentAPI.getEnchantmentAsItemOrNull(enchantmentId.substringBefore(":"), i)?.let { return@getOrPut it }
+        }
+
+        return@getOrPut null
```

Comment:
level parameter is null and returns the highest result in the RepoEnchantmentAPI why do we need this?